mesonbuild/dependencies/dev.py

```diff
@@ -532,7 +531,7 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
         m = self.env.machines[self.for_machine]
 
         if 'java' not in environment.coredata.compilers[self.for_machine]:
-            detect_compiler_for(environment, 'java', self.for_machine)
+            detect_compiler('java', environment, self.for_machine, 'java')
```

Comment:
```suggestion
            detect_compiler(self.name, environment, self.for_machine, 'java')
```

This function doesn't take the language twice, it takes the dependency name for logging and later the language to detect.